### PR TITLE
[SHELL32] DoGetProductType(): Check RegQueryValueExW() dwType

### PR DESCRIPTION
**Pull request by SergeGautherie**

### CODE REVIEW

File: dll/win32/shell32/wine/shellpath.c [hunk 1]
```diff
@@ -87,8 +87,8 @@ DoGetProductType(PNT_PRODUCT_TYPE ProductType)
         return FALSE;
 
     cbValue = sizeof(szValue);
-    error = RegQueryValueExW(hKey, L"ProductType", NULL, NULL, (LPBYTE)szValue, &cbValue);
-    if (!error)
+    error = RegQueryValueExW(hKey, L"ProductType", NULL, &dwType, (LPBYTE)szValue, &cbValue);
+    if (!error && dwType == REG_SZ)
     {
         if (lstrcmpW(szValue, L"WinNT") == 0)
             *ProductType = NtProductWinNt;
```
Comment (HBelusca) [inline dll/win32/shell32/wine/shellpath.c]: @katahiromz (for later): missing check for type NtProductServer.

Comment (SergeGautherie) [reply dll/win32/shell32/wine/shellpath.c]: `NtProductServer` is already set as default value...

Comment (HBelusca) [reply dll/win32/shell32/wine/shellpath.c]: Oh yes that's true.

File: dll/win32/shell32/wine/shellpath.c [hunk 2]
```diff
@@ -67,11 +67,11 @@ typedef enum _NT_PRODUCT_TYPE
...
-    static const WCHAR ProductOptions[] = L"System\\CurrentControlSet\\Control\\ProductOptions";
+    static const WCHAR ProductOptions[] = L"SYSTEM\\CurrentControlSet\\Control\\ProductOptions";
     HKEY hKey;
     LONG error;
     WCHAR szValue[32];
-    DWORD cbValue;
+    DWORD cbValue, dwType;
     static DWORD s_dwProductType = 0;
 
     if (s_dwProductType != 0)
```
Comment (HBelusca) [inline dll/win32/shell32/wine/shellpath.c]: So this means that if someone during runtime (but not by killing the shell) changes the product type to something else, the checked type will remain the old one?

**Pull request merged**